### PR TITLE
#13767: Remove ingestion pipeline action buttons and ingestion button if pipelineService is disabled

### LINKED ISSUE #13767: UI to disable ingestion pipeline actions if pipelineService is disabled
we have a `pipelineServiceClient.enabled: true/false` configuration. If it is not enabled, the UI should update the following:

- [ ] The run, logs, deploy,... actions from the UI should not appear.

We can still register pipelines only from YAML https://github.com/open-metadata/OpenMetadata/pull/13501

image

- [ ] We need to remove the `Add Metadata Ingestion` button. No ingestions can't be run from the UI without the pipeline service client
- [ ] The placeholder message should link [these docs](https://docs.open-metadata.org/v1.2.x/deployment/ingestion). Update the message with: `To view Ingestion Data, run the metadata ingestion on your own environment following these docs (link)`

---
# Development

To test:
1. Run the server with `PIPELINE_SERVICE_CLIENT_ENABLED=false`
2. The status call `http://localhost:8585/api/v1/services/ingestionPipelines/status` (which we are already using) will give you `{"code":200,"platform":"disabled","version":"disabled"}`

Thanks

### PR DESCRIPTION
**Pull request by Ashish8689**
…

### Describe your changes:

Fixes #13767

- Remove ingestion pipeline action buttons  pipeline service is disabled
- Remove the Add Metadata ingestion button and Add service button if pipeline service is disabled
- Change ingestion table message as per disable pipeline service

image

image

image

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx [hunk 1]
```diff
@@ -168,6 +178,7 @@ function IngestionListTable({
       handleIsConfirmationModalOpen,
       onIngestionWorkflowsUpdate,
       ingestionData,
+      isFetchingStatus,
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/IngestionListTable.component.tsx:181]: we also need to add `isPlatFormDisabled` in dependency?

File: openmetadata-ui/src/main/resources/ui/src/utils/IngestionUtils.tsx [hunk 2]
```diff
@@ -191,7 +192,10 @@ export const getIngestionTypes = (
   ];
 };
 
-const getPipelineExtraInfo = (pipelineType?: PipelineType) => {
+const getPipelineExtraInfo = (
+  isPlatFormDisabled: boolean,
+  pipelineType?: PipelineType
+) => {
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/utils/IngestionUtils.tsx:198]: as you are changing the position of the `pipelineType`, please check all the places where it is used

**Review by ShaileshParmar11 — COMMENTED** [review]
left few comments

File: openmetadata-ui/src/main/resources/ui/src/constants/docs.constants.ts [hunk 3]
```diff
@@ -27,6 +27,9 @@ export const CONNECTORS_DOCS = 'https://docs.open-metadata.org/connectors';
 export const WORKFLOWS_METADATA_DOCS =
   'https://docs.open-metadata.org/connectors/ingestion/workflows/metadata';
 
+export const INGESTION_FRAMEWORK_DEPLOYMENT_DOCS =
+  'https://docs.open-metadata.org/v1.2.x/deployment/ingestion';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/constants/docs.constants.ts:31]:
```suggestion
  'https://docs.open-metadata.org/deployment/ingestion';
```

**Comment by Ashish8689** [comment]
@ShaileshParmar11  @chirag-madlani changes made as per comments!

**Comment by ShaileshParmar11** [comment]
can you please take care of code smell?

**Comment by Ashish8689** [comment]
@ShaileshParmar11  the ingestion table code smell is handled!

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**